gradio/mcp.py

```diff
@@ -150,6 +141,14 @@ async def call_tool(
                 name: The name of the tool to call.
                 arguments: The arguments to pass to the tool.
             """
+            context_request = self.mcp_server.request_context.request
+            assert context_request is not None  # noqa: S101
+            root_url = route_utils.get_root_url(
+                request=context_request,
+                route_path="/gradio_api/mcp/messages",
```

Comment:
If using the http protocol, the path  `/gradio_api/mcp/http`, no? is this going to cause problems?

Reply:
good catch I'll fix this